llvm/test/Transforms/InstCombine/freeze.ll

```diff
@@ -877,6 +877,39 @@ exit:                                             ; preds = %loop
   ret void
 }
 
+; The recurrence for the GEP offset can't produce undef so the freeze should be
```

Comment:
```suggestion
; The recurrence for the GEP offset can't produce poison so the freeze should be
```
This will satisfy the checker and if more accurate. GEP will never produce undef, only poison. And on that note, you'll probably also want a test that *can* produce poison, by adding inbounds/nuw on the GEP.

Reply:
I've changed it to poison. For the other test, inbounds/nuw on the GEP is dropped and the freeze pushed thru: https://godbolt.org/z/r7aWf5nqq, unless `noundef` is removed from `%n`: https://godbolt.org/z/8j4Gx64o9. I can add a test for this.